### PR TITLE
add %newobject in neuralnet interface

### PR DESCRIPTION
**Pull request by sanuj**
For #3246. Fixes segfault while doing:
```
layers = NeuralLayers()
layers = layers.input(num_feats)
layers = layers.linear(50)
layers = layers.softmax(2)
```

### CODE REVIEW

**Comment by karlnapf** [comment]
Can you valgrind your mini example for leaks?

**Comment by lisitsyn** [comment]
:+1:

**Comment by sanuj** [comment]
[Valgrind output](https://gist.github.com/sanuj/052df069becd588315a2407e6ea274cc) using `valgrind --tool=memcheck --suppressions=valgrind-python.supp python ./feedforward_net_classification.py` by following [this](http://stackoverflow.com/questions/3982036/how-can-i-use-valgrind-with-python-c-extensions) link.

**Comment by karlnapf** [comment]
So this works now?

**Comment by lisitsyn** [comment]
One more thing!

Lets put a warning in the doc of these methods.

File: src/shogun/neuralnets/NeuralLayers.h [hunk 1]
```diff
@@ -47,6 +47,7 @@ namespace shogun
  * Uses builder-style syntax to describe layers
  * one by one.
  *
+ * While adding a new layer, also add it's prototype in SWIG with %newobject.
```
Comment (karlnapf) [inline src/shogun/neuralnets/NeuralLayers.h]: This both contains a grammar error "it's -> its", and also doesnt make any sense to me :)

rather remove, or replace with something meaningful. Recall that users of Shogun dont know what SWIG is. Rather say that every methods returns a reference to the same object or so

**Pull request merged**